tools/unsupervised_dataset/sql_classifier/parserc/src/main/java/Parserc.java

```diff
@@ -0,0 +1,156 @@
+import com.opencsv.*;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+
+import org.apache.calcite.sql.parser.dialect1.Dialect1ParserImpl;
+import org.apache.calcite.sql.parser.bigquery.BigQueryParserImpl;
+import org.apache.calcite.sql.parser.defaultdialect.DefaultDialectParserImpl;
+import org.apache.calcite.sql.parser.postgresql.PostgreSQLParserImpl;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Date;
+
+public class Parserc {
+    /*
+     * Runs the classification tool. Takes a CSV file of queries, classifies queries based on dialect, and creates
+     * several subdirectories to store the queries by dialect.
+     *
+     * @param args Command line arguments
+     */
+    public static void main(String[] args) {
+        if (args.length == 0) {
+            System.out.println("Please provide a CSV file.");
+            return;
+        }
+        List<String[]> allData = readCSV(args[0]);
+        if (allData == null) {
+            return;
+        }
+        SqlParser.Config[] parserConfigs = new SqlParser.Config[4];
+        parserConfigs[0] = SqlParser.configBuilder().setParserFactory(DefaultDialectParserImpl.FACTORY).build();
```

Comment:
Instead of adding it one by one, is it possible to do it in a loop to add all dialects Calcite has. Therefore, when Calcite has new dialect, it will be automatically included?

Reply:
I'm not sure if this is possible, especially with the workaround we are using since we cannot get some of the dependencies online yet? I think this would definitely make the most sense when we have the capabilities, so this can be a change once it is available.